manifests/repository/apt.pp

```diff
@@ -5,15 +5,28 @@
 # === Parameters
 #
 class curator::repository::apt (
+  String $debian_major_version = $::os['release']['major'],
   String $version = $::curator::repository::version,
 ) inherits curator::repository {
 
   if ! defined( Class['::apt'] ) {
     class { '::apt': }
   }
 
+  $repo_suffix = $debian_major_version ? {
+    # Debian releases
+    '8'     => '',
+    '9'     => '9',
```

Comment:
This is probably easier :
```
$repo_suffix = $::os['release']['major'] ? {
  '9'         => '9',
  default => '',
}
```

Reply:
I chose to do that way because Debian 10 will surely not work with the "debian" repo.  It _might_ work with the "debian9" repo, but maybe not :man_shrugging: 

It might be better to crash hard with unsupported systems, and add support for them accordingly when they are released?

Reply:
Ah ok, yes in that case let's leave the parameter `$debian_major_version` so that people can potentially use it with debian 10 before we patch the module 👍. However lets keep only *9* and *default* in the case statement

Reply:
Added in b41fee3cf51abc25453f5717728f40aa9b10b396

Please note that I am not endorsing this since the module will successfully deploy a broken setup, and because of the way the module works, the end-user will not be aware of the breakage (until he discovers that curator is non-functional because he ran out of disk space).

For the record, here is what the problem looks like when you hit it: https://github.com/elastic/curator/issues/1241

Reply:
if you prefer I'm happy to get what you used to have but make it fail on default then. Otherwise we just going to get an undef value. Might also need to get a conditional around it to only execute the switch case on Debian base hosts though.

Reply:
Hehe, when no default is present and no match is found, the selector fails (this is what I was relying on):

```
romain@zappy ~ % puppet apply -e '$x = $nobody_knows ? { 1 => "a", 2 => "b", } '                                                                            
Warning: Unknown variable: 'nobody_knows'. (line: 1, column: 6)
Error: Evaluation Error: No matching entry for selector parameter with value '' (line: 1, column: 6) on node zappy.home.sigabrt.org
```

You are right, a default with a better error message is probably the best!